src/main/java/org/openhab/binding/zwave/internal/protocol/commandclass/ZWaveMultiInstanceCommandClass.java

```diff
@@ -453,10 +453,34 @@ private void handleMultiChannelEncapResponse(SerialMessage serialMessage, int of
         int originatingEndpointId = serialMessage.getMessagePayloadByte(offset);
         int destinationEndpointId = serialMessage.getMessagePayloadByte(offset + 1);
 
-        if (useDestEndpointAsSource) {
-            // Not a full swap. Do not use destinationEndpointId after this line
-            // and leave scope intact.
-            originatingEndpointId = destinationEndpointId;
+        // if a specific endpoint on the controller is addressed.
+        if (destinationEndpointId > 1) {
+            // swap specified in node options and condition satisfied:
+            if (useDestEndpointAsSource && originatingEndpointId <= 1) {
+                // swap originating and destination.
+                int temp = originatingEndpointId;
+                originatingEndpointId = destinationEndpointId;
+                destinationEndpointId = temp;
+            } else {
+                // received an encapsulation for an endpoint on the controller.
+                logger.info(
+                        "NODE {}: Received a multi instance encapsulation with a destination endpoint = {}. "
+                                + "The controller in our binding does not have endpoints.",
```

Comment:
Can we reduce the log message to one line - the controller might have endpoints one day so we shouldn't make this statement.